### ISSUE TITLE
Regexp java 8 issue

### ISSUE THREAD

**Issue by clausfod** [checkstyle/checkstyle#3002]
I get a false positive for this line of code:
```
import java.util.Date;
import java.util.List;

public class ItemService {

    public void getAlreadyExistsReasonCode(Item item, List<Item> items) {
        Date minPublicFrom = items.stream().map(Item::getPublicFrom).min(Date::compareTo).get();
    }
}
```
with this checkstyle.xml:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://jee-qa/jee-qa/trunk/checkstyle/configuration_1_2.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Regexp">
            <property name="format"
                value="User:|Date:|Time:|Created by IntelliJ IDEA.|Copyright " />
            <property name="illegalPattern" value="true" />
            <property name="message"
                value=" User:|Date:|Time:|Created by IntelliJ IDEA.|Copyright are autogenerated comments - are noise - remove it" />
        </module>
    </module>
</module>
```
java -jar checkstyle-6.16.1-all.jar -c checkstyle.xml ItemService.java

Results in this:
Line matches the illegal pattern ' User:|Date:|Time:|Created by IntelliJ IDEA.|Copyright are autogenerated comments - are noise - remove it'. [Regexp]

**Comment by romani** [comment]
@clausfod , why you think this is false positive. It is your responsibility to make Regular Expression work without false positives.
One of the main reason for us to have special Checks is because parsing by regexp is extremely unstable and full of fase-positives. Regexp Check is only a tool with its own limitations by design.

Advice: make your regexp more complicated to avoid such false positives.
Looks like that you want to control java comments content by regexp, but there is no such Check for now, but you can make it ,let me know it want to create it, it is so simple.

**Comment by clausfod** [comment]
Hi Romani,

Thanks for the reply. We have millions lines of code where there is no problem with this Regexp. Only positives are reported. 

After upgrading to Checkstyle 6.x (I will find the version) this started to happen - so I guess there has been a change in Checkstyle concerning this. I will return monday with a version in which the problem started.

Have a nice weekend.
Kind regards
Claus Nielsen

**Comment by romani** [comment]
> We have millions lines of code where there is no problem with this Regexp

welcome to nightmare of parsing by regexp.

please use https://regex101.com/ to recheck your line of code against your regexp.

**Comment by som-snytt** [comment]
One of our rules was the `System.out` example but omitted `ignoreComments`. I complained, because I often comment out debug printlns (instead of promoting them to logging). It seems to me this issue wants an `ignoreCode` property for `RegExpSingleLineJava`.

I'll volunteer that fix if that sounds good to you.

**Comment by MEZk** [comment]
@clausfod 
It seams to me that your regexp is a bit wrong. If you want to force the check to validate appearing of specific text in comments, you should use something like this:

^(\/\/|\*|\/\*\*|\/\*).*(User|Date|Time|Created by IntelliJ IDEA\.|Copyright):.*

**Comment by romani** [comment]
> It seems to me this issue wants an ignoreCode property for RegExpSingleLineJava.

I do not like this idea. RegExpSingleLineJava should stay with java code validation only. 
We need to make new Check to validate comments by RegExp. There are already talk about such Checks, but looks like issue is not created yet but have to be create.
Nuance: right now chekcstyle in moratorium period([reason](https://groups.google.com/forum/#!topic/checkstyle-devel/JbhfiiSqKsQ)) for any new Checks, I do not accept them, all new Checks are welcome through our experimental project - [sevntu.checkstyle](https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues).   Please open issue there on this idea.

> I'll volunteer that fix if that sounds good to you.

you are welcome, I will guide you through development process. See you in sevntu.checkstyle.
